internal/expr/prepare.go

```diff
@@ -88,15 +88,24 @@ func prepareInput(ti typeNameToInfo, p *inputPart) (tm typeMember, err error) {
 		}
 	}()
 
-	info, ok := ti[p.sourceType.typeName]
+	info, ok := ti[p.sourceType.TypeName()]
 	if !ok {
-		return nil, typeMissingError(p.sourceType.typeName, getKeys(ti))
+		return nil, typeMissingError(p.sourceType.TypeName(), getKeys(ti))
 	}
 
-	tm, err = info.typeMember(p.sourceType.memberName)
-	if err != nil {
-		return nil, err
+	switch t := p.sourceType.(type) {
+	case keyedAccesor:
+		if t.memberName == "*" {
+			return nil, fmt.Errorf(`asterisk used with %s in invalid context`, info.typ().Kind())
+		}
+		tm, err = info.typeMember(t.memberName)
+		if err != nil {
+			return nil, err
+		}
+	case sliceRangeAccessor:
+		return nil, fmt.Errorf("internal error: slice support not implemented")
```

Comment:
imo this doesn't need the internal error prefix, this is something we want to tell the user if they encounter it.

Reply:
True. FYI, at the moment we never reach this code because prepare errors out if you pass it a slice.

Reply:
Ah yea, good point.